opensrp-pnc/src/main/java/org/smartregister/pnc/processor/PncMiniClientProcessorForJava.java

```diff
@@ -48,12 +67,185 @@ public boolean canProcess(@NonNull String eventType) {
 
     @Override
     public void processEventClient(@NonNull EventClient eventClient, @NonNull List<Event> unsyncEvents, @Nullable ClientClassification clientClassification) throws Exception {
-        // do nothing
+
+        /*if (eventType.equals(PncConstants.EventTypeConstants.PNC_REGISTRATION)
+                || eventType.equals(PncConstants.EventTypeConstants.UPDATE_PNC_REGISTRATION)) {
+            processEvent(eventClient.getEvent(), eventClient.getClient(), clientClassification);
+            CoreLibrary.getInstance().context().getEventClientRepository().markEventAsProcessed(eventClient.getEvent().getFormSubmissionId());
+        } else if (eventType.equals(PncConstants.EventTypeConstants.PNC_CLOSE)) {
+            if (eventClient.getClient() == null) {
+                throw new PncCloseEventProcessException(String.format("Client %s referenced by %s event does not exist", event.getBaseEntityId(), PncConstants.EventTypeConstants.PNC_CLOSE));
+            }
+            processEvent(eventClient.getEvent(), eventClient.getClient(), clientClassification);
+            CoreLibrary.getInstance().context().getEventClientRepository().markEventAsProcessed(eventClient.getEvent().getFormSubmissionId());
+            unsyncEvents.add(event);
+        } else if (eventType.equals(PncConstants.EventTypeConstants.PNC_OUTCOME)) {
+            processEvent(eventClient.getEvent(), eventClient.getClient(), clientClassification);
+            processPncOutcome(event);
+            CoreLibrary.getInstance().context().getEventClientRepository().markEventAsProcessed(eventClient.getEvent().getFormSubmissionId());
+        }*/
+
+
+        Event event = eventClient.getEvent();
+
+        String eventType = event.getEventType();
+
+        if (eventType.equals(PncConstants.EventTypeConstants.PNC_REGISTRATION)
+                || eventType.equals(PncConstants.EventTypeConstants.UPDATE_PNC_REGISTRATION)) {
+            ArrayList<EventClient> eventClients = new ArrayList<>();
+            eventClients.add(eventClient);
+            processClient(eventClients);
+
+            //updateRegisterTypeColumn(event, "maternity");
+
+            HashMap<String, String> keyValues = new HashMap<>();
+            generateKeyValuesFromEvent(event, keyValues, true);
+
+            PncRegistrationDetails maternityDetails = new PncRegistrationDetails(eventClient.getClient().getBaseEntityId(), event.getEventDate().toDate(), keyValues);
+            maternityDetails.setCreatedAt(new Date());
+
+            PncLibrary.getInstance().getPncRegistrationDetailsRepository().saveOrUpdate(maternityDetails);
+//            processEvent(eventClient.getEvent(), eventClient.getClient(), clientClassification);
+//            CoreLibrary.getInstance().context().getEventClientRepository().markEventAsProcessed(eventClient.getEvent().getFormSubmissionId());
```

Comment:
Could you update this logic to be as that of maternity .... just on the registration and update registration events bit. Then clean up any additional code

Reply:
I have updated the file according to the need in issue 11 branch